src/Compilers/VisualBasic/Test/Semantic/Semantics/NonTrailingNamedArgumentsTests.vb

```diff
@@ -0,0 +1,832 @@
+                                          ~
+BC30241: Named argument expected. Please use language version 15.6 or greater to use non-trailing named arguments.
+        System.Console.Write(P(index1:=1, 0 to 5))
+                                          ~
+BC32017: Comma, ')', or a valid expression continuation expected.
+        System.Console.Write(P(index1:=1, 0 to 5))
+                                            ~~~~
+</expected>)
+        End Sub
+
+        <Fact>
+        Public Sub TestInIndexer2()
+
+            Dim compilationDef =
+<compilation>
+    <file name="a.vb">
+Class C
+    Public ReadOnly Property P(ByVal index1 As Integer, Optional ByVal index2 As Integer = -1) As String
+        Get
+            Return System.String.Format($"{index1} {index2}. ")
+        End Get
+    End Property
+    Sub M()
+        System.Console.Write(P(index1:=1, 2))
+        System.Console.Write(P(index1:=3, ))
+        System.Console.Write(P(index1:=4, index2:=5))
+        System.Console.Write(P(index2:=7, index1:=6))
+    End Sub
+    Shared Sub Main()
+        Dim c = New C()
+        c.M()
+    End Sub
+End Class
+    </file>
+</compilation>
+
+            Dim compilation = CompilationUtils.CreateCompilationWithMscorlibAndVBRuntime(compilationDef,
+                parseOptions:=TestOptions.Regular.WithLanguageVersion(LanguageVersion.Latest), options:=TestOptions.DebugExe)
+
+            CompilationUtils.AssertNoDiagnostics(compilation)
+            CompileAndVerify(compilation, expectedOutput:="1 2. 3 -1. 4 5. 6 7.")
+        End Sub
+
+        <Fact>
+        Public Sub TestInConstructor()
+
+            Dim compilationDef =
+<compilation>
+    <file name="a.vb">
+Class C
+    Public Sub New(input1 As Integer, Optional input2 As Integer = 2, Optional input3 As Integer = 3)
+    End Sub
+    Shared Sub Main()
+        Dim c = New C(input1:=1, , bad:=3)
+    End Sub
+End Class
+    </file>
+</compilation>
+
+            Dim compilation = CompilationUtils.CreateCompilationWithMscorlibAndVBRuntime(compilationDef,
+                parseOptions:=TestOptions.Regular.WithLanguageVersion(LanguageVersion.Latest), options:=TestOptions.DebugExe)
+
+            CompilationUtils.AssertTheseDiagnostics(compilation, <errors>
+BC30272: 'bad' is not a parameter of 'Public Sub New(input1 As Integer, [input2 As Integer = 2], [input3 As Integer = 3])'.
+        Dim c = New C(input1:=1, , bad:=3)
+                                   ~~~
+                                                                 </errors>)
+        End Sub
+
+        <Fact>
+        Public Sub TestInConstructor2()
+
+            Dim compilationDef =
+<compilation>
+    <file name="a.vb">
+Class C
+    Public Sub New(input1 As Integer, Optional input2 As Integer = 2, Optional input3 As Integer = 3)
+        System.Console.Write($"{input1} {input2} {input3}. ")
+    End Sub
+    Shared Sub Main()
+        Dim c = New C(input1:=1, , 3)
+        c = New C(input1:=1, , 0 to 4)
+    End Sub
```

Comment:
>End Sub [](start = 4, length = 7)

It would be good to test success cases when an omitted argument or a range argument is in between named arguments.